packages/scaffold/src/partials/w3m-connecting-wc-mobile/index.ts

```diff
@@ -36,8 +41,15 @@ export class W3mConnectingWcMobile extends W3mConnectingWidget {
     if (this.wallet?.mobile_link && this.uri) {
       try {
         this.error = false
-        const { mobile_link, name } = this.wallet
-        const { redirect, href } = CoreHelperUtil.formatNativeUrl(mobile_link, this.uri)
+        const { mobile_link: deep_link, link_mode: universal_link, name } = this.wallet
+        let link = deep_link
+        if (OptionsController.state.enableUniversalLinks) {
+          if (!universal_link) {
+            throw new Error('Universal Link is undefined.')
```

Comment:
should we throw an error here? or just use the deeplink as fallback

Reply:
The flag is meant to use only Universal Link, in this case because Telegram doesn't support Deep Linking, so it would be the same for this particular case. Generally it is prefered Deeplinking and if it's undefined it will fallback to Universal Link in the API.

I added this error message because it's possible that a wallet supports DeppLinking but not Universal Link in which case with an error message it's easier to debug and we'll know what the issue is right away in case there are any reports.

So for general use cases this flag shouldn't be used only edge cases like the Telegram Mini Apps one.

Reply:
mm okey. This solution is kind of a patch, the really good solution would be that if `enableUniversalLinks`, we should only show wallets compatible with universal links

Reply:
Yeah sounds good! As for now we only need this as a Canary so I guess we could add this

Reply:
should be filtered now